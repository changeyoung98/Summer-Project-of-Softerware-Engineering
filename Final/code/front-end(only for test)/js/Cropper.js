import React from 'react';
import Cropper from 'react-cropper';
import 'cropperjs/dist/cropper.css';
import {Button,Input,Icon} from 'antd'
/* global FileReader */

const src = 'child.jpg';
export default class Crop extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            src,
            cropResult: null,
        };
        this.cropImage = this.cropImage.bind(this);
        this.onChange = this.onChange.bind(this);
        this.useDefaultImage = this.useDefaultImage.bind(this);
        this.load = this.load.bind(this);
    }

    onChange(e) {
        e.preventDefault();
        let files;
        if (e.dataTransfer) {
            files = e.dataTransfer.files;
        } else if (e.target) {
            files = e.target.files;
        }
        const reader = new FileReader();
        reader.onload = () => {
            this.setState({ src: reader.result });
        };
        reader.readAsDataURL(files[0]);
    }

    cropImage() {
        if (typeof this.cropper.getCroppedCanvas() === 'undefined') {
            return;
        }
        this.setState({
            cropResult: this.cropper.getCroppedCanvas().toDataURL(),
        });
    }

    useDefaultImage() {
        this.setState({ src });
    }

    load(){
        document.getElementById('upfile').click()
    }

    render() {
        return (
            <div>
                <div style={{ width: '100%' }}>
                    <div className="fileInput left">
                        <input type="file" name="upfile" id="upfile" className="upfile" onChange={this.onChange}/>
                        <Button  onClick={this.load}><Icon type="folder-open" />Select File</Button>
                    </div>
                    <div>
                    <Button onClick={this.useDefaultImage}>Use default img</Button>
                    </div>
                    <br />
                    <br />
                    <Cropper
                        style={{ height: 400, width: '100%' }}
                       // aspectRatio={9 / 16}
                        preview=".img-preview"
                        guides={false}
                        src={this.state.src}
                        ref={cropper => { this.cropper = cropper; }}
                    />
                </div>
                <div>
                    <div className="box" style={{ width: '50%', float: 'right' }}>
                        <h1>
                            <span>
                            Preview
                            </span>
                           </h1>
                        <div className="img-preview" style={{ width: '100%', float: 'left', height: 500 }} />
                    </div>
                    <div className="box" style={{ width: '50%', float: 'right' }}>
                        <h1>
                            <div>
                            <span>Crop</span>
                            <Button onClick={this.cropImage} style={{ float: 'right' }}>
                                Crop Image
                            </Button>
                            <Button  style={{ float: 'right' }}><Icon type="upload" />Upload</Button>
                            </div>
                        </h1>
                        <img style={{ width: '100%' }} src={this.state.cropResult} alt="cropped image" />
                    </div>
                </div>
                <br style={{ clear: 'both' }} />
            </div>
        );
    }
}/**
 * Created by hp on 2018/7/11.
 *  <!--  <Input type="file" style={{opacity: 0}} onChange={this.onChange} /> -->
 */
