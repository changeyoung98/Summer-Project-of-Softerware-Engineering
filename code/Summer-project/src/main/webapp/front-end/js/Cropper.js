import React from 'react';
import Cropper from 'react-cropper';
import 'cropperjs/dist/cropper.css';
import {Button,Input,Icon} from 'antd'
/* global FileReader */

const src = 'child.jpg';
export default class Crop extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            src,
            cropResult: null,
        };
        this.cropImage = this.cropImage.bind(this);
        this.onChange = this.onChange.bind(this);
        this.useDefaultImage = this.useDefaultImage.bind(this);
    }

    onChange(e) {
        e.preventDefault();
        let files;
        if (e.dataTransfer) {
            files = e.dataTransfer.files;
        } else if (e.target) {
            files = e.target.files;
        }
        const reader = new FileReader();
        reader.onload = () => {
            this.setState({ src: reader.result });
        };
        reader.readAsDataURL(files[0]);
    }

    cropImage() {
        if (typeof this.cropper.getCroppedCanvas() === 'undefined') {
            return;
        }
        this.setState({
            cropResult: this.cropper.getCroppedCanvas().toDataURL(),
        });
    }

    useDefaultImage() {
        this.setState({ src });
    }

    render() {
        return (
            <div>
                <div style={{ width: '100%' }}>
                    <div >
                    <Input type="file"  onChange={this.onChange} >
                         Select File
                    </Input>
                    </div>
                    <div>
                    <Button onClick={this.useDefaultImage}>Use default img</Button>
                    </div>
                    <br />
                    <br />
                    <Cropper
                        style={{ height: 400, width: '100%' }}
                        aspectRatio={9 / 16}
                        preview=".img-preview"
                        guides={false}
                        src={this.state.src}
                        ref={cropper => { this.cropper = cropper; }}
                    />
                </div>
                <div>
                    <div className="box" style={{ width: '50%', float: 'right' }}>
                        <h1>
                            <span>
                            Preview
                            </span>
                           </h1>
                        <div className="img-preview" style={{ width: '100%', float: 'left', height: 500 }} />
                    </div>
                    <div className="box" style={{ width: '50%', float: 'right' }}>
                        <h1>
                            <span>Crop</span>
                            <div>
                            <Button onClick={this.cropImage} style={{ float: 'right' }}>
                                Crop Image
                            </Button>
                            <Button  style={{ float: 'right' }}><Icon type="upload" />Upload</Button>
                            </div>
                        </h1>
                        <img style={{ width: '50%' }} src={this.state.cropResult} alt="cropped image" />
                    </div>
                </div>
                <br style={{ clear: 'both' }} />
            </div>
        );
    }
}/**
 * Created by hp on 2018/7/11.
 */
