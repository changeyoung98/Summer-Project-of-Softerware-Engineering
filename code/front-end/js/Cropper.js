import React from 'react';
import Cropper from 'react-cropper';
import 'cropperjs/dist/cropper.css';
import {Button,Input} from 'antd'
/* global FileReader */

const src = 'child.jpg';

export default class Crop extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            src,
            cropResult: null,
        };
        this.cropImage = this.cropImage.bind(this);
        this.onChange = this.onChange.bind(this);
        this.useDefaultImage = this.useDefaultImage.bind(this);
    }

    onChange(e) {
        e.preventDefault();
        let files;
        if (e.dataTransfer) {
            files = e.dataTransfer.files;
        } else if (e.target) {
            files = e.target.files;
        }
        const reader = new FileReader();
        reader.onload = () => {
            this.setState({ src: reader.result });
        };
        reader.readAsDataURL(files[0]);
    }

    cropImage() {
        if (typeof this.cropper.getCroppedCanvas() === 'undefined') {
            return;
        }
        this.setState({
            cropResult: this.cropper.getCroppedCanvas().toDataURL(),
        });
    }

    useDefaultImage() {
        this.setState({ src });
    }

    render() {
        return (
            <div>
                <div style={{ width: '100%' }}>
                    <div className="file-box">
                    <Input type="file" className="file-btn" onChange={this.onChange} />
                    </div>
                    <Button type="primary" onClick={this.useDefaultImage}>Use default img</Button>
                    <br />
                    <br />
                    <Cropper
                        style={{ height: 400, width: '100%' }}
                        aspectRatio={16 / 9}
                        preview=".img-preview"
                        guides={false}
                        src={this.state.src}
                        ref={cropper => { this.cropper = cropper; }}
                    />
                </div>
                <div>
                    <div className="box" style={{ width: '50%', float: 'right' }}>
                        <h1>Preview</h1>
                        <div className="img-preview" style={{ width: '100%', float: 'left', height: 300 }} />
                    </div>
                    <div className="box" style={{ width: '50%', float: 'right' }}>
                        <h1>
                            <span>Crop</span>
                            <Button onClick={this.cropImage} style={{ float: 'right' }}>
                                Crop Image
                            </Button>
                        </h1>
                        <img style={{ width: '100%' }} src={this.state.cropResult} alt="cropped image" />
                    </div>
                </div>
                <br style={{ clear: 'both' }} />
            </div>
        );
    }
}/**
 * Created by hp on 2018/7/11.
 */
