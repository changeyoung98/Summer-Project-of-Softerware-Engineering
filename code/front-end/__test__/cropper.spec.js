/**
 * Created by hp on 2018/7/19.
 */
import React from 'react'
import Cropper from '../js/Cropper'
import {Button,Input,Icon} from 'antd'
import toJson from 'enzyme-to-json';
import { shallow, configure,render } from 'enzyme'
import  Adapter from 'enzyme-adapter-react-14';

configure({ adapter: new Adapter() });
const setup = () => {
    // 模拟 props
    const src = 'child.jpg';
    const props = {
        // Jest 提供的mock 函数
        state : {
        src,
        cropResult: null,
    },
    cropImage : jest.fn(),
    onChange :jest.fn(),
    useDefaultImage :jest.fn(),
    load :jest.fn()
    };
    const wrapper = shallow(<Cropper />)
    return {
        props,
        wrapper
    }
}
describe('Cropper render', () => {
   const {wrapper,props}=setup();
    it('basic use', () => {
        const wrapper = render(
            <Cropper style={{ height: 400, width: '100%' }} />
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
        it('Button Component should render', () => {
            //.find(selector) 是 Enzyme shallow Rendering 提供的语法, 用于查找节点
            // 详细用法见 Enzyme 文档 http://airbnb.io/enzyme/docs/api/shallow.html
            expect(wrapper.find('Button').length).toBe(4);
        });
    it('Input Component should render', () => {
        //.find(selector) 是 Enzyme shallow Rendering 提供的语法, 用于查找节点
        // 详细用法见 Enzyme 文档 http://airbnb.io/enzyme/docs/api/shallow.html
        expect(wrapper.find('input').length).toBe(1);
    });
    it('image Component should render', () => {
        //.find(selector) 是 Enzyme shallow Rendering 提供的语法, 用于查找节点
        // 详细用法见 Enzyme 文档 http://airbnb.io/enzyme/docs/api/shallow.html
        expect(wrapper.find('img').length).toBe(1);
    });
});