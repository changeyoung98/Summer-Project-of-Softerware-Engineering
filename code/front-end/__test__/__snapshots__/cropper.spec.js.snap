// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cropper render basic use 1`] = `
<div>
  <div
    style="width:100%;"
  >
    <div
      class="fileInput left"
    >
      <input
        class="upfile"
        id="upfile"
        name="upfile"
        type="file"
      />
      <button
        class="ant-btn"
        type="button"
      >
        <i
          class="anticon anticon-folder-open"
        />
        <span>
          Select File
        </span>
      </button>
    </div>
    <div>
      <button
        class="ant-btn"
        type="button"
      >
        <span>
          Use default img
        </span>
      </button>
    </div>
    <br />
    <br />
    <div
      style="height:400px;width:100%;"
    >
      <img
        alt="picture"
        src="child.jpg"
        style="opacity:0;"
      />
    </div>
  </div>
  <div>
    <div
      class="box"
      style="width:50%;float:right;"
    >
      <h1>
        <span>
          Preview
        </span>
      </h1>
      <div
        class="img-preview"
        style="width:100%;float:left;height:500px;"
      />
    </div>
    <div
      class="box"
      style="width:50%;float:right;"
    >
      <h1>
        <div>
          <span>
            Crop
          </span>
          <button
            class="ant-btn"
            style="float:right;"
            type="button"
          >
            <span>
              Crop Image
            </span>
          </button>
          <button
            class="ant-btn"
            style="float:right;"
            type="button"
          >
            <i
              class="anticon anticon-upload"
            />
            <span>
              Upload
            </span>
          </button>
        </div>
      </h1>
      <img
        alt="cropped image"
        style="width:100%;"
      />
    </div>
  </div>
  <br
    style="clear:both;"
  />
</div>
`;
